crates/kornia-imgproc/src/filter/separable_filter.rs

```diff
@@ -88,6 +88,72 @@ pub fn separable_filter<const C: usize>(
     Ok(())
 }
 
+
+/// Apply a fast filter horizontally, take advantage of property where all
+/// weights are equal to filter an image at O(n) speed
+/// 
+/// # Arguments
+///
+/// * `src` - The source image with shape (H, W, C).
+/// * `dst_transposed` - The destination image with shape (W, H, C).
+/// * `half_kernel_x_size` - Half of the kernel at weight 1. The total size would be 2*this+1
+pub fn fast_horizontal_filter<const C: usize>(
+    src: &Image<f32, C>,
+    dst_transposed: &mut Image<f32, C>,
+    half_kernel_x_size: usize,
+) -> Result<(), ImageError> {
+    let src_data = src.as_slice();
+    let dst_transposed_data = dst_transposed.as_slice_mut();
+    let mut row_acc = [0.0; C];
+    
+    let mut leftmost_pixel = [0.0; C];
+    let mut rightmost_pixel = [0.0; C];
+    for (pix_offset, source_pixel) in src_data.iter().enumerate() {
+        let ch = pix_offset % C;
+        let rc = pix_offset / C;
+        let c = rc % src.cols();
+        let r = rc / src.cols();
+
+        let transposed_r = c;
+        let transposed_c = r;
+        let transposed_pix_offset = transposed_r*src.rows()*C + transposed_c*C + ch;
+
+        if c == 0 {
+            row_acc[ch] = *source_pixel * (half_kernel_x_size+1) as f32;
+            let mut kernel_pix_offset = pix_offset;
```

Comment:
Wondering wether this offset could be precomputed beforehand as you are computing several times in the top level function